src/main/java/edu/harvard/iq/dataverse/engine/command/impl/FinalizeDatasetPublicationCommand.java

```diff
@@ -256,30 +256,23 @@ public boolean onSuccess(CommandContext ctxt, Object r) {
             }
         }
 
-        exportMetadata(dataset);
-                
-        ctxt.datasets().updateLastExportTimeStamp(dataset.getId());
-
-        return retVal;
-    }
-
-    /**
-     * Attempting to run metadata export, for all the formats for which we have
-     * metadata Exporters.
-     */
-    private void exportMetadata(Dataset dataset) {
-
+        // Metadata export:
+        
         try {
             ExportService instance = ExportService.getInstance();
             instance.exportAllFormats(dataset);
-
+            ctxt.datasets().updateLastExportTimeStamp(dataset.getId());
```

Comment:
Why both this and setting the update time in exportAllFormats? 

Could/should it just be removed here? (Or is a ctxt.datasets().merge(dataset); or similar needed to get the update from the ExportService? )

Can you even get rid of updateLastExportTimeStamp? (I see it called three places, all after a call to exportAllFormats (andexportAllFormats is called a few more places where updateLastExportTimeStamp isn't called)).

Reply:
I'm a bit stumped by this actually... With the line 264 above commented out the time stamp does not get updated. And I can't immediately tell why. Is there still something finicky about the transaction under which onSuccess() is called, in this context, of FinalizeDatasetPublicationCommand being executed as a "nested" command? 
Not sure about .merge(dataset), why it would be needed - as opposed to having all the changes done to the dataset getting saved automatically at the end of the transaction. ExportService is not an EJB itself, despite the name. 
OK, I would need to experiment some more to figure this out.

Reply:
Uhm, I would like to just declare this out of scope - figuring out why that line is necessary. It has been there for a while.

Reply:
I'd suggest at least a comment since you found it isn't doing anything / doesn't appear to anyway.

Reply:
Actually, an explicit .merge() call there makes calling that method unnecessary.
Guessing it may be possible to get rid of it in the other 2 places.

Reply:
ok.